src/diff/children.js

```diff
@@ -169,27 +171,27 @@ export function diffChildren(parentDom, newParentVNode, oldParentVNode, context,
 }
 
 /**
- * Flatten a virtual nodes children to a single dimensional array
+ * Flatten and loop through the children of a virtual node
  * @param {import('../index').ComponentChildren} children The unflattened
  * children of a virtual node
- * @param {Array<import('../internal').VNode | null>} [flattened] An flat array of children to modify
- * @param {typeof import('../create-element').coerceToVNode} [map] Function that
- * will be applied on each child if the `vnode` is not `null`
- * @param {boolean} [keepHoles] wether to coerce `undefined` to `null` or not.
- * This is needed for Components without children like `<Foo />`.
+ * @param {(vnode: import('../internal').VNode) => import('../internal').VNode} [callback]
+ * A function to invoke for each child before it is added to the flattened list.
+ * @param {import('../internal').VNode[]} [flattened] An flat array of children to modify
+ * @returns {import('../internal').VNode[]}
  */
-export function toChildArray(children, flattened, map, keepHoles) {
+export function toChildArray(children, callback, flattened) {
```

Comment:
IIRC the first two args here are used externally. Maybe worth changing to `toChildArray(children, flattened, callback)`?

Reply:
I think that change was made for golfing purposes in `children.js`. There it's used without the `flattened` argument. Either way we've only advertised the first arg so far in our types and elsewhere. So I think it's fine to swap them like it's done here.